arch/riscv/Kconfig

```diff
@@ -119,6 +119,22 @@ config RISCV_SOC_HAS_CUSTOM_SYS_IO
 	  the RISC-V SoC needs to do something different and more than reading and
 	  writing the registers.
 
+config RISCV_SOC_HAS_CUSTOM_SMP_IPI
```

Comment:
The current implementation is using CLINT, I wonder if we should name this `RISCV_SOC_SMP_IPI_CLINT` instead to pave the way for the intro of `RISCV_SOC_SMP_IPI_PLIC` in the future, so that it is a choice between CLINT & PLIC, instead of default vs custom?

Reply:
It would make sense if we upstream our implementation as the default PLIC implementation, otherwise it would be unclear as to what the behavior would be if RISCV_SOC_SMP_IPI_CLINT was not set

Reply:
I think it should be fine as this is an arch-level Kconfig, would like to hear from @fkokosinski @npitre and others as well